### PR TITLE
Backfill rspec tests for UpdateDeployment 

### PR DESCRIPTION
**Pull request by jaresty**
Previous description:

> Before this change, no matter what config IDs were being passed into
UpdateDeployment, it was using an empty array for the cloud config
models and runtime config models when preparing the update deployments.
This change adds test coverage for that code path and fixes the loading
of those models at the beginning of UpdateDeployment#perform.

Updated:
This PR backfills unit tests that were missing from the UpdateDeployment unit tests.

### CODE REVIEW

File: src/bosh-director/lib/bosh/director/jobs/update_deployment.rb [hunk 1]
```diff
@@ -35,15 +35,15 @@ def perform
           @event_log.warn_deprecated(warning)
           cloud_config_models = nil
         else
-          cloud_config_models = Bosh::Director::Models::Config.find_by_ids(@cloud_config_ids)
+          cloud_config_models = Bosh::Director::Models::Config.where(id: @cloud_config_ids).to_a
```
Comment (jfmyers9) [inline src/bosh-director/lib/bosh/director/jobs/update_deployment.rb]: Is there a reason to change this from using `find_by_ids`? From the definition on the class it seems to be doing a very similar thing.

Comment (jaresty) [reply src/bosh-director/lib/bosh/director/jobs/update_deployment.rb]: I thought it should be working as is, but the tests showed it was not actually working.  Without this change, I was unable to get the `cloud_config_ids` to result in cloud_config objects passed to the test doubles.

Comment (jaresty) [reply src/bosh-director/lib/bosh/director/jobs/update_deployment.rb]: I see the issue now - `find_by_ids` is actually a custom find method defined on our model (I had assumed it was provided by Sequel).  I'll investigate a bit further to see why it's not finding the models we're creating for test.

Comment (jaresty) [reply src/bosh-director/lib/bosh/director/jobs/update_deployment.rb]: We were stubbing its return value in the test.  I removed the stub and restored the original code.  Now this PR is just adding some new test coverage.

**Comment by jaresty** [comment]
@jfmyers9 I backfilled tests for the orphaned network cleanup and saw some results that seemed surprising.  It's possible that I made a mistake in my test setup, but it looks like we are removing unmanaged networks if they are on the deployment model (where being on means that the network names match). That behavior sounds incorrect to me, but maybe I'm wrong.

**Comment by jfmyers9** [comment]
@jaresty Do you think it's valuable to break this out into a chore or a bug to explore further so that we can get a pair to spend some time digging into it during working hours? Otherwise I'll remind the community pair to take a look at this tomorrow.

**Pull request closed without merging**